src/NovelRT/Windowing/Glfw/GlfwWindowingDevice.cpp

```diff
@@ -71,9 +74,12 @@ namespace NovelRT::Windowing::Glfw
         auto extensions = glfwGetRequiredInstanceExtensions(&extensionCount);
         if (extensionCount == 0)
         {
-            const char* output = nullptr;
-            glfwGetError(&output);
-            throw Exceptions::InitialisationFailureException("GLFW3 failed to initialise.", std::string(output));
+           const char* output = nullptr;
+           glfwGetError(&output);
+           if(output != nullptr)
+           {
+                throw Exceptions::InitialisationFailureException("GLFW3 failed to initialise.", std::string(output));
+           }
```

Comment:
This is missing the additional branch I asked for - I've appended a suggestion you can apply. This slightly differs from what I originally wanted in terms of the actual message, but it saves adding an additional ctor overload if it doesn't exist and semantically makes more sense given how Vulkan works.

```suggestion
           if(output != nullptr)
           {
                throw Exceptions::InitialisationFailureException("GLFW3 failed to initialise.", std::string(output));
           }
           else
           {
               throw Exceptions::InitialisationFailureException("GLFW3 failed to initialise.", "Attempting to fetch the required Vulkan extensions failed with a count of zero.");
           }
```